core/src/types/address.rs

```diff
@@ -111,8 +112,10 @@ pub enum DecodeError {
     UnexpectedBech32Prefix(String, String),
     #[error("Unexpected Bech32m variant {0:?}, expected {BECH32M_VARIANT:?}")]
     UnexpectedBech32Variant(bech32::Variant),
+    #[error("Invalid address encoding: {0}, {1}")]
+    InvalidInnerEncoding(ErrorKind, String),
     #[error("Invalid address encoding")]
-    InvalidInnerEncoding(std::io::Error),
+    InvalidInnerEncodingStr(String),
```

Comment:
I think you can remove the `std::io::ErrorKind` too as we don't need to differentiate between them (the strings should capture what the issue is already)

Reply:
Probably, but here it's not too bad, and it's like appending more information on, in a way.